sources/packages/backend/apps/api/src/auth/_tests_/e2e/maintenance-mode.guard.e2e-spec.ts

```diff
@@ -0,0 +1,266 @@
+    app.init();
+    const dataSource = moduleFixture.get(DataSource);
+    db = createE2EDataSources(dataSource);
+    configServiceMockHelper = new ConfigServiceMockHelper(app);
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  describe("AllowDuringMaintenanceMode decorator", () => {
+    it("Should allow @AllowDuringMaintenanceMode route when global maintenance is ON", async () => {
+      configServiceMockHelper.setMaintenanceMode({ maintenanceMode: true });
+      await request(app.getHttpServer()).get("/config").expect(HttpStatus.OK);
+    });
+
+    it("Should allow @Public route when all maintenance flags are ON", async () => {
+      configServiceMockHelper.setMaintenanceMode({
+        maintenanceMode: false,
+        maintenanceModeStudent: true,
+        maintenanceModeInstitution: true,
+        maintenanceModeMinistry: true,
+        maintenanceModeSupportingUser: true,
+        maintenanceModeExternal: true,
+      });
+      await request(app.getHttpServer())
+        .get("/auth-test/public-route")
+        .expect(HttpStatus.OK);
+    });
+  });
+
+  describe("Student maintenance mode", () => {
+    it("Should block Student when MAINTENANCE_MODE_STUDENT is ON", async () => {
+      configServiceMockHelper.setMaintenanceMode({
+        maintenanceModeStudent: true,
+      });
+      const studentToken = await getStudentToken(
+        FakeStudentUsersTypes.FakeStudentUserType1,
+      );
+      await request(app.getHttpServer())
+        .get("/auth-test/user-not-required-route")
+        .auth(studentToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.SERVICE_UNAVAILABLE)
+        .expect({
+          statusCode: HttpStatus.SERVICE_UNAVAILABLE,
+          message:
+            "Service temporarily unavailable for maintenance. Please try again later or contact support if the issue persists.",
+          error: "Service Unavailable",
+        });
+    });
+
+    it("Should allow Student when MAINTENANCE_MODE_STUDENT is OFF", async () => {
+      configServiceMockHelper.setMaintenanceMode({
+        maintenanceModeStudent: false,
+      });
+      const studentToken = await getStudentToken(
+        FakeStudentUsersTypes.FakeStudentUserType1,
+      );
+
+      await request(app.getHttpServer())
+        .get("/auth-test/user-not-required-route")
+        .auth(studentToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK);
+    });
+  });
+
+  describe("Institution maintenance mode", () => {
+    it("Should block Institution when MAINTENANCE_MODE_INSTITUTION is ON", async () => {
+      configServiceMockHelper.setMaintenanceMode({
+        maintenanceModeInstitution: true,
+      });
+      const institutionToken = await getInstitutionToken(
+        InstitutionTokenTypes.CollegeCUser,
+      );
+      const collegeELocation = await getAuthorizedLocation(
+        db,
+        InstitutionTokenTypes.CollegeEReadOnlyUser,
+        InstitutionUserTypes.readOnlyUser,
+      );
```

Comment:
No need to have a proper location for this validation.
The endpoint can be harcoded as
`const endpoint = `/auth-test/institution-location-reading-route/999999`;`

Reply:
updated